src/components/candevice/canfactory.hpp

```diff
@@ -1,12 +1,20 @@
 #ifndef CANFACTORY_HPP_C394TY5U
 #define CANFACTORY_HPP_C394TY5U
 
-#include <QtCore/QString>
+#if QT_VERSION >= QT_VERSION_CHECK(5, 9, 1)
+#include <QCanBusDeviceInfo>
+#endif
+
+#include <QList>
+#include <QString>
 
 struct CanDeviceInterface;
 struct CanFactoryInterface {
     virtual ~CanFactoryInterface() {}
     virtual CanDeviceInterface* create(const QString& backend, const QString& iface) = 0;
+#if QT_VERSION >= QT_VERSION_CHECK(5, 9, 1)
+    virtual QList<QCanBusDeviceInfo> availableDevices(const QString& backend, QString* errorMessage = nullptr) const = 0;
```

Comment:
The Qt implementation of relevant method is not marked as noexcept, also the documentation doesn't mention any guarantees in this respect:
https://doc.qt.io/qt-5/qcanbus.html#availableDevices
I can mark it noexcept, but it might be misleading.